CONTRIBUTING.md

```diff
@@ -0,0 +1,140 @@
+    ```
+
+    You should now be able to see the branch on GitHub. Make commits as you deem necessary. It helps to provide useful commit messages - a commit message saying 'Update' is a lot less helpful than saying 'Remove X parameter because it was unused'.
+
+    ```shell
+    git commit changed_file.txt -m "Remove X parameter because it was unused"
+    git push
+    ```
+
+1. (Make sure you have follow instructions in "Install development dependencies") Once you have made your additions or changes, make sure you write tests and run the test suite.  More information on testing below.
+
+    ```shell
+    pytest -vs tests/
+    ```
+
+1. (Make sure you have follow instructions in "Install development dependencies") Make sure to run the auto python code formatter, black.
+
+    ```shell
+    black ./
+    ```
+
+1. Once you have completed all the steps above, create a pull request from the feature branch to the `dev` branch of the Sage-Bionetworks/agora-data-tools repo.
+
+> *A code maintainer must review and accept your pull request.* A code review ideally happens with both the contributor and the reviewer present, but is not strictly required for contributing. This can be performed remotely (e.g., Zoom, Hangout, or other video or phone conference).  Most code reviews are done asyncronously.
+
+This package uses [semantic versioning](https://semver.org/) for releasing new versions. A github release should occur at least once a quarter to capture the changes between releases.
+
+<!-- This package uses [semantic versioning](https://semver.org/) for releasing new versions. The version should be updated on the `dev` branch as changes are reviewed and merged in by a code maintainer. The version for the package is maintained in the [agoradatatools/__init__.py](agoradatatools/__init__.py) file.  A github release should also occur every time `dev` is pushed into `main` and it should match the version for the package. -->
+
+### Testing
+
+#### Running tests
+
+This package uses [`pytest`](https://pytest.org/en/latest/) to run tests. The test code is located in the [tests](./tests) subdirectory.
+
+Here's how to run the test suite:
+
+```shell
+pytest -vs tests/
+```
+
+Tests are also run automatically by Github Actions on any pull request and are required to pass before merging.
+
+#### Test Development
+
+Please add tests for new code. These might include unit tests (to test specific functionality of code that was added to support fixing the bug or feature), integration tests (to test that the feature is usable - e.g., it should have complete the expected behavior as reported in the feature request or bug report), or both.
+
+##### Mock Testing
+
+It is recommended to use the following style (see example below) for mock testing across this package:
+
+```python
+from unittest.mock import patch
+...
+patch.object(MODULE_NAME, "FUNCTION_TO_MOCK_NAME".return_value=SOME_RETURN_VALUE)
+```
+
+<!-- ### Release Procedure (For Package Maintainers)
```

Comment:
Are all the commented out bits targeted at package maintainers (or some other specific) audience? If so, it might be good to mention that up front either globally if all the same audience, or per chunk if not.

Reply:
I think its a bit confusing when you are reading the raw markdown, but you won't see it when you click on the rendered markdown: https://github.com/Sage-Bionetworks/agora-data-tools/blob/tyu/add-contributing/CONTRIBUTING.md.

This section is commented out, because it hasn't been fully established.